### ISSUE TITLE
samples: hello-world: extra slash in path

### ISSUE THREAD

**Issue by eugkr** [zephyrproject-rtos/zephyr#31791]
Please help with zephyr setup, looks like something misconfigured.
Project successfully compiles, but shows error on flash:
```
PS E:\eug\zephyrproject\zephyr\samples\hello_world> west build -b nucleo_f411re
-- west build: generating a build system
Including boilerplate (Zephyr base): E:/Eug/zephyrproject/zephyr/cmake/app/boilerplate.cmake
-- Zephyr version: 2.5.0-rc1 (E:/Eug/zephyrproject/zephyr)
-- Found Python3: C:/Python/python.exe (found suitable exact version "3.8.2") found components: Interpreter
-- Found west (found suitable version "0.8.0", minimum required is "0.7.1")
-- Board: nucleo_f411re
-- Found dtc: C:/ProgramData/chocolatey/bin/dtc.exe (found suitable version "1.4.7", minimum required is "1.4.6")
-- Found toolchain: gnuarmemb (C:/Program Files (x86)/GNUARM)
-- Found BOARD.dts: E:/Eug/zephyrproject/zephyr/boards/arm/nucleo_f411re/nucleo_f411re.dts
-- Generated zephyr.dts: E:/Eug/zephyrproject/zephyr/samples/hello_world/build/zephyr/zephyr.dts
-- Generated devicetree_unfixed.h: E:/Eug/zephyrproject/zephyr/samples/hello_world/build/zephyr/include/generated/devicetree_unfixed.h
-- Generated device_extern.h: E:/Eug/zephyrproject/zephyr/samples/hello_world/build/zephyr/include/generated/device_extern.h
Parsing E:/Eug/zephyrproject/zephyr/Kconfig
Loaded configuration 'E:/Eug/zephyrproject/zephyr/boards/arm/nucleo_f411re/nucleo_f411re_defconfig'
Merged configuration 'E:/Eug/zephyrproject/zephyr/samples/hello_world/prj.conf'
Configuration saved to 'E:/Eug/zephyrproject/zephyr/samples/hello_world/build/zephyr/.config'
Kconfig header saved to 'E:/Eug/zephyrproject/zephyr/samples/hello_world/build/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 10.2.1
-- The CXX compiler identification is GNU 10.2.1
-- The ASM compiler identification is GNU
-- Found assembler: C:/Program Files (x86)/GNUARM/bin/arm-none-eabi-gcc.exe
CMake Deprecation Warning at E:/Eug/zephyrproject/modules/lib/civetweb/CMakeLists.txt:2 (cmake_minimum_required):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Configuring done
-- Generating done
-- Build files have been written to: E:/Eug/zephyrproject/zephyr/samples/hello_world/build
-- west build: building application
[127/132] Linking C executable zephyr\zephyr_prebuilt.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       14396 B       512 KB      2.75%
            SRAM:        4384 B       128 KB      3.34%
        IDT_LIST:         184 B         2 KB      8.98%
[132/132] Linking C executable zephyr\zephyr.elf
PS E:\eug\zephyrproject\zephyr\samples\hello_world> west flash
-- west flash: rebuilding
ninja: error: FindFirstFileExA(e/:/eug/zephyrproject/zephyr/misc): The filename, directory name, or volume label syntax is incorrect.

FATAL ERROR: re-build in E:\eug\zephyrproject\zephyr\samples\hello_world\build failed (no --build-dir given)
PS E:\eug\zephyrproject\zephyr\samples\hello_world> git rev-parse --short HEAD
30de4b5dd9
PS E:\eug\zephyrproject\zephyr\samples\hello_world> cmake --version
cmake version 3.19.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
PS E:\eug\zephyrproject\zephyr\samples\hello_world> arm-none-eabi-gcc --version
arm-none-eabi-gcc.exe (GNU Arm Embedded Toolchain 10-2020-q4-major) 10.2.1 20201103 (release)
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
PS E:\eug\zephyrproject\zephyr\samples\hello_world> ls env:
TOOLCHAIN_DIR                  C:\Program Files (x86)\GNUARM
ZEPHYR_BASE                    E:/Eug/zephyrproject/zephyr
ZEPHYR_TOOLCHAIN_VARIANT       gnuarmemb
```
In file build/.ninja_deps all paths contain extra slash after drive letter:  e/:/eug/zephyrproject/zephyr/misc
Checked on another computer - no slash after drive letter.
Also: "CMake Deprecation Warning" not appeared earlier

System: Windows 10 x64

Any advice please?

**Comment by eugkr** [comment]
Looks like magic.
Both machines have the same windows build, the same versions of GCC, cmake, west, ninja, I did clean zephyr install with upgrading all packets from requirements.txt, but no success:
```
ninja: error: FindFirstFileExA(e/:/eug/zephyrproject/zephyr/samples/hello_world/build/zephyr/include/generated): The filename, directory name, or volume label syntax is incorrect.  
FATAL ERROR: command exited with status 1: 'C:\Program Files\CMake\bin\cmake.EXE' --build 'E:\Eug\zephyrproject\zephyr\samples\hello_world\build'
```

**Comment by eugkr** [comment]
Fixed by updating ninja from 1.10.0 to 1.10.2
This bug mentioned in ninja release notes.
